Saule/Resources/DefaultApiResourceProviderFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Net.Http;
+
+namespace Saule.Resources
+{
+    /// <summary>
+    /// Default provider factory that resolves attributes based on attributes and HttpRequestMessage
+    /// </summary>
+    public class DefaultApiResourceProviderFactory : IApiResourceProviderFactory
+    {
+        /// <inheritdoc/>
+        public IApiResourceProvider Create(HttpRequestMessage request)
+        {
+            object resource;
+            if (request.Properties.TryGetValue(Constants.PropertyNames.ResourceDescriptor, out resource) && resource is ApiResource)
+            {
+                var apiResource = (ApiResource)resource;
```

Comment:
```suggestion
            if (request.Properties.TryGetValue(Constants.PropertyNames.ResourceDescriptor, out var resource) && resource is ApiResource apiResource)
```

Reply:
My initial code version had almost the same code - https://github.com/BookerSoftwareInc/saule/commit/0c166fd661a8d86014e059cc9e9f97d987c75391#diff-76813ea3f0bb77be87bd0ea4a4540059R13 but looks like AppVeyor uses older language version and it failed for me with compilation errors and i had to replace it to older syntax - https://ci.appveyor.com/project/JoukevanderMaas/saule/builds/34704891/messages

we can probably update AppVeyor to use 7.3 language version like csproj has and uses as locally that initial code compiles fine but i'm not totally sure how to do that

Reply:
I think I need to move to azure devops at some point, I'll do that eventually but for now it's fine to ignore this then.